actioncable/lib/action_cable/channel/base.rb

```diff
@@ -235,11 +249,7 @@ def subscribe_to_channel
             subscribed
           end
 
-          if subscription_rejected?
-            reject_subscription
-          else
-            transmit_subscription_confirmation unless defer_subscription_confirmation?
-          end
+          reject_subscription if subscription_rejected?
```

Comment:
We probably need to avoid `transmit_subscription_confirmation` if `subscription_rejected?`... otherwise it seems we could transmit a confirmation *after* we've transmitted a rejection.